src/backends/backend/triton_model_instance.cc

```diff
@@ -69,6 +70,10 @@ TritonModelInstance::CreateInstances(
     TritonModel* model, const inference::ModelConfig& model_config)
 {
   for (const auto& group : model_config.instance_group()) {
+    std::set<std::string> profile_names;
```

Comment:
The order of the profile names is important since you access them by index and because we should present them in the same order as they appear in the model configuration, so you should use a vector here. You can't rely on the ordering in a set.